checks.d/mapreduce.py

```diff
@@ -0,0 +1,451 @@
+
+# Metrics to collect
+MAPREDUCE_JOB_METRICS = {
+    'elapsedTime': ('mapreduce.job.elapsed_time', HISTOGRAM),
+    'mapsTotal': ('mapreduce.job.maps_total', HISTOGRAM),
+    'mapsCompleted': ('mapreduce.job.maps_completed', HISTOGRAM),
+    'reducesTotal': ('mapreduce.job.reduces_total', HISTOGRAM),
+    'reducesCompleted': ('mapreduce.job.reduces_completed', HISTOGRAM),
+    'mapsPending': ('mapreduce.job.maps_pending', HISTOGRAM),
+    'mapsRunning': ('mapreduce.job.maps_running', HISTOGRAM),
+    'reducesPending': ('mapreduce.job.reduces_pending', HISTOGRAM),
+    'reducesRunning': ('mapreduce.job.reduces_running', HISTOGRAM),
+    'newReduceAttempts': ('mapreduce.job.new_reduce_attempts', HISTOGRAM),
+    'runningReduceAttempts': ('mapreduce.job.running_reduce_attempts', HISTOGRAM),
+    'failedReduceAttempts': ('mapreduce.job.failed_reduce_attempts', HISTOGRAM),
+    'killedReduceAttempts': ('mapreduce.job.killed_reduce_attempts', HISTOGRAM),
+    'successfulReduceAttempts': ('mapreduce.job.successful_reduce_attempts', HISTOGRAM),
+    'newMapAttempts': ('mapreduce.job.new_map_attempts', HISTOGRAM),
+    'runningMapAttempts': ('mapreduce.job.running_map_attempts', HISTOGRAM),
+    'failedMapAttempts': ('mapreduce.job.failed_map_attempts', HISTOGRAM),
+    'killedMapAttempts': ('mapreduce.job.killed_map_attempts', HISTOGRAM),
+    'successfulMapAttempts': ('mapreduce.job.successful_map_attempts', HISTOGRAM),
+}
+
+MAPREDUCE_JOB_COUNTER_METRICS = {
+    'reduceCounterValue': ('mapreduce.job.counter.reduce_counter_value', HISTOGRAM),
+    'mapCounterValue': ('mapreduce.job.counter.map_counter_value', HISTOGRAM),
+    'totalCounterValue': ('mapreduce.job.counter.total_counter_value', HISTOGRAM),
+}
+
+MAPREDUCE_MAP_TASK_METRICS = {
+    'progress': ('mapreduce.job.map.task.progress', HISTOGRAM),
+    'elapsedTime': ('mapreduce.job.map.task.elapsed_time', HISTOGRAM)
+}
+
+MAPREDUCE_REDUCE_TASK_METRICS = {
+    'progress': ('mapreduce.job.reduce.task.progress', HISTOGRAM),
+    'elapsedTime': ('mapreduce.job.reduce.task.elapsed_time', HISTOGRAM)
+}
+
+class MapReduceCheck(AgentCheck):
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
+
+        # Parse any custom counter metrics
+        self.custom_counter_metrics = self._parse_custom_metrics(init_config)
+
+    def check(self, instance):
+        # Get properties from conf file
+        rm_address = instance.get('resourcemanager_uri')
+        if rm_address is None:
+            raise Exception('The ResourceManager URL must be specified in the instance configuration')
+
+        # Get the cluster ID from Yarn
+        cluster_id = self._get_cluster_id(rm_address)
+
+        # Get the running MR applications from YARN
+        running_apps = self._get_running_app_ids(rm_address,
+            states=YARN_APPLICATION_STATES,
+            applicationTypes=YARN_APPLICATION_TYPES)
```

Comment:
Nitpick: coud you move the usage of these parameters further down, inside the `_get_running_app_ids` method, on the `_rest_request_to_json` function call?